src/Compilers/Core/Portable/CommandLine/ErrorLogger.cs

```diff
@@ -272,7 +275,8 @@ private void WriteValue(Value value, bool isFirst = true, bool valueInList = fal
 
         private void WriteValue(string value)
         {
-            _writer.Write($"\"{value}\"");
+            _writer.Flush();
+            s_jsonStringSerializer.WriteObject(_writer.BaseStream, value);
```

Comment:
It is not exception safe, but the values output to the error log file should be generated by the compiler, so shouldn't cause an exception.

Thread safety: You are correct it is not thread safe, and can be a problem in VBCSCompiler mode. I'll switch this to an instance field, and also associate the instance to the writer itself (as per your last review comment).